packages/gatsby-remark-autolink-headers/src/index.js

```diff
@@ -20,13 +20,30 @@ module.exports = (
     className = `anchor`,
     maintainCase = false,
     removeAccents = false,
+    enableCustomId = false,
   }
 ) => {
   slugs.reset()
 
   visit(markdownAST, `heading`, node => {
-    const slug = slugs.slug(toString(node), maintainCase)
-    const id = removeAccents ? deburr(slug) : slug
+    let id
+    if (enableCustomId && node.children.length > 0) {
+      const last = node.children[node.children.length - 1]
+      const match = /^(.*?)\s*\{#([\w-]+)\}$/.exec(toString(last))
```

Comment:
could you add a comment above this line that tells us what this regex does?
Something like: `// Matches {#<word>} at the end of a string

Reply:
Done